### PR TITLE
Features/only my messages

### LINKED ISSUE #7326: Commit dialog: Show only my messages
## Feature description

Please add a possibility to show only my commit messages in commit message popup.

## Environment

- Git Extensions 3.2.1.6628
- Build bc07663c3c893f1d896520e0bdcfbe35cfe4b729
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4018.0
- DPI 96dpi (no scaling)

### PR DESCRIPTION
**Pull request by idealist1508**
Fixes #7326 

## Proposed changes

- GitCommands 
  - GetPreviousCommitMessages allows filter by author
- GitUi
  - Commit dialog: New option to show only my messages.

### Before

image

### After
image

## Test methodology 

- Manual test 

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by pmiossec** [comment]
I would have preferred to see the option at the bottom of the menu (because you don't have to change it often)

File: GitCommands/Settings/AppSettings.cs [hunk 1]
```diff
@@ -368,6 +368,12 @@ public static int CommitDialogNumberOfPreviousMessages
             set => SetInt("commitDialogNumberOfPreviousMessages", value);
         }
 
+        public static bool CommitDialogShowOnlyMyMessages
+        {
+            get => GetBool("commitDialogShowOnlyMyMessages", false);
```
Comment (pmiossec) [inline GitCommands/Settings/AppSettings.cs]: I prefer a default value of 'true'.
what do you all think about it?

Comment (gerhardol) [reply GitCommands/Settings/AppSettings.cs]: I prefer false, to not hide messages by default

File: GitUI/CommandsDialogs/FormCommit.Designer.cs [hunk 2]
```diff
@@ -184,9 +185,9 @@ private void InitializeComponent()
             this.toolbarCommit.SuspendLayout();
             this.commitStatusStrip.SuspendLayout();
             this.SuspendLayout();
-            //
+            //
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormCommit.Designer.cs]: Why was extra spaces added here?
VS may remove that at next update creating noise...
It is often preferred to not apply such changes.

File: GitUI/CommandsDialogs/FormCommit.Designer.cs [hunk 3]
```diff
@@ -213,185 +214,192 @@ private void InitializeComponent()
             this.toolStripSeparator4,
             this.interactiveAddToolStripMenuItem});
             this.UnstagedFileContext.Name = "UnstagedFileContext";
-            this.UnstagedFileContext.Size = new System.Drawing.Size(233, 414);
-            this.UnstagedFileContext.Opening += UnstagedFileContext_Opening;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormCommit.Designer.cs]: Why was this removed?

File: GitUI/CommandsDialogs/FormCommit.Designer.cs [hunk 4]
```diff
@@ -406,88 +414,87 @@ private void InitializeComponent()
             this.stagedEditFileToolStripMenuItem11});
             this.StagedFileContext.Name = "StagedFileContext";
             this.StagedFileContext.Size = new System.Drawing.Size(233, 198);
-            this.StagedFileContext.Opening += StagedFileContext_Opening;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormCommit.Designer.cs]: Also removed

Comment (idealist1508) [reply GitUI/CommandsDialogs/FormCommit.Designer.cs]: I can't explain it. I made revert.

**Comment by idealist1508** [comment]
> Rebase and squash before merge
I never done it before. I hope i done it right.

**Comment by gerhardol** [comment]
> I never done it before. I hope i done it right.

You squashed the commits, but merged master into the feature branch instead of rebasing, so not quite.
If you just hard reset the branch to before the merge it is almost OK, but you should also rebase on master.
(You can also do this in one step by resetting mixed to upstream/master, committing and push force-with-lease)

**Comment by gerhardol** [comment]
@msftbot merge after 48 hours

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 5]
```diff
@@ -2267,9 +2269,18 @@ private void CommitMessageToolStripMenuItemDropDownOpening(object sender, EventA
         {
             var msg = AppSettings.LastCommitMessage;
             var maxCount = AppSettings.CommitDialogNumberOfPreviousMessages;
+            var authorPattern = "";
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormCommit.cs]:
```suggestion
            var authorPattern = string.Empty;
```

File: contributors.txt [hunk 6]
```diff
@@ -102,4 +102,5 @@ YYYY/MM/DD, github id, Full name, email
 2019/10/13, bansan85, Le Garrec Vincent, legarrec.vincent{at]gmail.com
 2019/10/14, FabMan08, Fabrizio Mancin, fabman08(at)gmail.com
 2019/10/15, manishkungwani, Manish Kungwani, manishkungwani{at]yahoo.com
-2019/10/18, lhiginbotham, Logan Higinbotham, logan.higinbotham(at)gmail.com
\ No newline at end of file
+2019/10/18, lhiginbotham, Logan Higinbotham, logan.higinbotham(at)gmail.com
+2019/10/18, idealist1508, none, none(at)none.de
```
Comment (RussKie) [inline contributors.txt]: ❗️  You must provide your real name and your real email

Please re-read the document

File: GitCommands/Git/GitModule.cs [hunk 7]
```diff
@@ -3523,6 +3523,12 @@ public IEnumerable<string> GetPreviousCommitMessages(int count, string revision
                 revision,
                 "--pretty=format:%e%n%s%n%n%b"
             };
+
+            if (!string.IsNullOrEmpty(authorPattern))
+            {
+                args.Add($"--author=\"{authorPattern}\"");
+            }
```
Comment (RussKie) [inline GitCommands/Git/GitModule.cs]: Please merge the check into the `GitArgumentBuilder` initialisation logic above
```cs
            var args = new GitArgumentBuilder("log")
            {
                "-z",
                $"-n {count}",
                revision,
                "--pretty=format:%e%n%s%n%n%b",
                { !string.IsNullOrEmpty(authorPattern), string.Concat("--author=\"", authorPattern, "\"") }
            };
```

**Comment by RussKie** [comment]
@msftbot ignore

**Comment by RussKie** [comment]
@msftbot forget everything I just told you

File: contributors.txt (hunk 6 above)
Comment (idealist1508) [reply contributors.txt]: I am sorry, but i don't wont to publish my real name and real email publicly. This is not acceptable for me.

File: GitUI/CommandsDialogs/FormCommit.cs (hunk 5 above)
Comment (idealist1508) [reply GitUI/CommandsDialogs/FormCommit.cs]: Half the file uses ```""``` and the other ```string.Empty```. Should I apply a action to the whole file?

File: contributors.txt (hunk 6 above)
Comment (gerhardol) [reply contributors.txt]: You could add a separate GitHub account (which I believe is OK from GitHub's point of view), with an email address separate from your normal. But it should be a real monitored account, not a fake one.

File: GitUI/CommandsDialogs/FormCommit.cs (hunk 5 above)
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormCommit.cs]: This comment was for what you added
